### ISSUE TITLE
Allow to parameterize the long press duration to enter drag mode

### ISSUE THREAD

**Issue by gdurandrexel** [igorkurilenko/animated_reorderable#2]
In my App, to edit the order of a list of items, the user switches to an "edit mode" in the UI. When in this mode, there is no reason to delay the drag start of an item.
I would like to be able to set this delay to zero in this case.

**Comment by tanguymossion** [comment]
I also would like the possibility to have a faster drag start

**Comment by igorkurilenko** [comment]
Thank you for your requests. I will add them shortly.

**Comment by igorkurilenko** [comment]
I've added `reorderGestureRecognizerFactory` to `AnimatedReorderable`. 

```dart
  /// A factory function that creates a [MultiDragGestureRecognizer] for handling
  /// reordering gestures in the [AnimatedReorderable] widget.
  ///
  /// By default, it uses [DelayedMultiDragGestureRecognizer], which introduces a delay
  /// to detect a long press before initiating the drag. This behavior is useful in cases where
  /// you want to distinguish between simple taps and drag gestures.
  ///
  /// Users can override this factory to:
  /// - Customize the default delay for initiating drag gestures (e.g., changing the delay duration).
  /// - Use different types of gesture recognizers (e.g., for specific user interactions or input devices).
  ///
  /// Default implementation:
  /// ```dart
  /// MultiDragGestureRecognizer createReoderDragGestureRecognizer(
  ///   BuildContext context,
  /// ) =>
  ///   DelayedMultiDragGestureRecognizer()
  ///     ..gestureSettings = MediaQuery.maybeGestureSettingsOf(context);
  /// ```
  ///
  /// Example of customizing the delay duration to 250ms using [DelayedMultiDragGestureRecognizer]:
  /// ```dart
  /// reoderGestureRecognizerFactory: (BuildContext context) {
  ///   final recognizer = DelayedMultiDragGestureRecognizer()
  ///     ..gestureSettings = MediaQuery.maybeGestureSettingsOf(context);
  ///
  ///   // Customize the delay duration to 250ms
  ///   recognizer.delay = const Duration(milliseconds: 250);
  ///
  ///   return recognizer;
  /// }
  /// ```
  ///
  /// This factory allows for fine-tuned control over the gesture recognition system,
  /// providing flexibility for various user interactions and devices.
  final MultiDragGestureRecognizerFactory? reoderGestureRecognizerFactory;
```

https://github.com/igorkurilenko/animated_reorderable/commit/1669b083f86cd849ba8f52ddd955d3a85b4f20a3

@tanguymossion @gdurandrexel 
Does this solution work for you?

**Comment by igorkurilenko** [comment]
> In my App, to edit the order of a list of items, the user switches to an "edit mode" in the UI. When in this mode, there is no reason to delay the drag start of an item. I would like to be able to set this delay to zero in this case.

@gdurandrexel 
You can now define `reorderGestureRecognizerFactory` to return `ImmediateMultiDragGestureRecognizer`

**Comment by gdurandrexel** [comment]
sounds great. Thanks.

**Comment by gdurandrexel** [comment]
damn, just found a typo in the new parameter name reoderGestureRecognizerFactory: missing R in reoRder

**Comment by igorkurilenko** [comment]
fixed in 1.4.2
